src/components/diff-settings.jsx

```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+
+/**
+ * @typedef DiffSettingsProps
+ * @property {string} diffType
+ * @property {Function} onRemoveFormattingChange
+ * @property {boolean} removeFormatting
+ */
+
+/**
+ *
+ * @class DiffSettings
+ * @extends {React.Component}
+ * @param {DiffSettingsProps} props
+
+ */
+export default class DiffSettings extends React.Component {
+  render () {
+    const handleChange = (event) => {
+      this.props.onRemoveFormattingChange(event.target.checked);
+    };
```

Comment:
Two little optimizations you can do here:
1. Extend `React.PureComponent`.
2. Pull this handler out into the class so we don’t need to recreate it on every render.